File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP2_sofiane/Concessionnaire.php
<?php
class Concessionnaire{

    // attributs

    private $nom;
    private static $id = 0;
    private $idConcessionnaire;
    private $catalogue;

    // constructeur par initialisation

    public function __construct($catalogue){
    	$this->idConcessionnaire = self::$id;
        Concessionnaire::$id++;
        $this->nom = "VoitureVendeur";
        $this->catalogue = $catalogue;
    }
    // getter

    public function getNom(){
        return $this->nom;
    }

    public function getIdConcessionnaire(){
        return $this->idConcessionnaire;
    }

    public function getCatalogue(){
        return $this->catalogue;
    }


    // setter

    public function setNom($nom){
        $this->nom = $nom;
    }

    public function setCatalogue($catalogue){
        $this->catalogue = $catalogue;
    }

    // methode toString

    public function __toString(){
        return "Concessionnaire [nom=" . $this->getNom() . ", idConcessionnaire=" . $this->getIdConcessionnaire() . ", catalogue=" . $this->catalogue. "]";
    }
}

File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP1-2/src/Concessionnaire.java
import constructeurAuto.Catalogue;

public class Concessionnaire {
	private String nom;
	private static int id = 0;
	private int idConcessionnaire;
	private Catalogue catalogue;
	
	public Concessionnaire(String nom, Catalogue catalogue) {
		this.idConcessionnaire = id;
		Concessionnaire.id ++;
		this.nom = nom;
		this.catalogue = catalogue;
	}
	
	public Concessionnaire(Catalogue catalogue){
		this.idConcessionnaire = id;
		Concessionnaire.id ++;
		this.nom = "VoitureVendeur";
		this.catalogue = catalogue;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public int getIdConcessionnaire() {
		return idConcessionnaire;
	}
	
	public Catalogue getCatalogue() {
		return this.catalogue;
	}
	
	public void setCatalogue(Catalogue catalogue) {
		this.catalogue = catalogue;
	}

	@Override
	public String toString() {
		return "Concessionnaire [nom=" + nom + ", idConcessionnaire=" + idConcessionnaire + ", catalogue=" + catalogue + "]";
	}
}
